_posts/2023-12-20-in-progress-wildfly-feature-on-openshift.adoc

```diff
@@ -0,0 +1,312 @@
+
+* https://docs.docker.com/[Docker] (you can also use https://docs.podman.io/en/latest/[Podman])  
+* https://developers.redhat.com/developer-sandbox[Developer Sandbox]
+* https://docs.openshift.com/container-platform/4.14/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI]
+* https://quay.io/repository/[Quay]
+* https://helm.sh/[Helm]
+
+== Example Application
+For this walkthrough, we will be making use of the https://github.com/PrarthonaPaul/elytron-examples/tree/elytron-oidc-client-scope/elytron-oidc-client-scope[elytron-oidc-client-scope] project which makes use of the ability to configure additional scope values for OpenID Connect. First, clone the `elytron-example` repo locally and navigate to the `elytron-oidc-client-scope` project:
+```
+git clone https://github.com/prarthonapaul/elytron-examples
+git checkout elytron-oidc-client-scope
+cd PATH/TO/elytron-examples/elytron-oidc-client-scope
+```
+
+== Log Into the OpenShift Cluster
+
+Before we can deploy our application, we need to log into an OpenShift cluster. You can log in via the OpenShift CLI:
+```
+oc login -u myUserName
+```
+
+Alternatively, to log in using the API token, navigate to Developer Sandbox, click on your username at the top right corner, and click on `Copy login command`. It will redirect you to a page where you can copy your login command and paste it on your terminal. 
+The command will be in this format: 
+```
+oc login --token=myToken --server=myServerUrl
+```
+
+Once you have your environment set up with the required tools, we can move on to the next step to build and deploy our application on OpenShift. 
+
+== Start Red Hat Single Sign On 
+Developer Sandbox gives us access to the enterprise version of Red Hat's OpenID provider: Red Hat Single Sign On (RH-SSO). On the topology view of OpenShift, click on the _Add to Project_ button at the top left and add the version of RH-SSO that is compatible with the version of OpehShift you are running and wait for it to finish deploying. Please refer to the https://access.redhat.com/documentation/en-us/red_hat_single_sign-on/7.6/html/red_hat_single_sign-on_for_openshift/index[RH-SSO for OpenShift Documentation] to learn more about it.
+
+**Note:** If the SSO pod fails to build with the message `ERROR *** JBossAS wrapper process (1) received TERM signal ***` on the terminal, then try increasing `timeoutSeconds` and `initialDelaySeconds` for both the `livenessProbe` and the `readinessProbe` for the SSO pod. For more information, visit https://access.redhat.com/solutions/4874991[here].
```

Comment:
Yes, I did run into this earlier in November. It was a bug with the RH-SSO pod. I think it may have been resolved, because the last time I ran through this guide, I did not run into this issue.